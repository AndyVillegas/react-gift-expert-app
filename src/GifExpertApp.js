import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({defaultCategories=[]}) => {
    const [categories, setCategories] = useState(defaultCategories);
    const handleClearCategory = ()=>{
        setCategories([]);
    }
    return (
        <>
            <h2>GifExpertApp</h2>
            <button className="btn-clear" onClick={handleClearCategory}>Limpiar b√∫squedas</button>
            <hr />
            <AddCategory setCategories={setCategories} />
            <div>
                {categories.map(category => (
                    <GifGrid
                        key={category}
                        category={category}
                    />))}
            </div>
        </>
    )
}

export default GifExpertApp
