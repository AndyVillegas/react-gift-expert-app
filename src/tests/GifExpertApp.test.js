import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
describe('Pruebas en GifExpertApp', () => {
    test('debe de renderizar correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['Naruto', 'Dragon Ball'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
});
