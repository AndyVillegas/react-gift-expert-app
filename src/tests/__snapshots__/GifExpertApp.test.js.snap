// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe de mostrar una lista de categorías 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <button
    className="btn-clear"
    onClick={[Function]}
  >
    Limpiar búsquedas
  </button>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <div>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe de renderizar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <button
    className="btn-clear"
    onClick={[Function]}
  >
    Limpiar búsquedas
  </button>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <div />
</Fragment>
`;
